plugin/trino-influxdb/src/main/java/io/trino/plugin/influxdb/client/InfluxClient.java

```
@@ -0,0 +1,356 @@
+                .connectTimeout(config.getConnectTimeOut().toMillis(), MILLISECONDS)
+                .writeTimeout(config.getWriteTimeOut().toMillis(), MILLISECONDS)
+                .readTimeout(config.getReadTimeOut().toMillis(), MILLISECONDS)
+                .addNetworkInterceptor(new HttpLoggingInterceptor(System.err::println));
+        this.client = createClient(config, httpBuilder);
+
+        if (config.getCacheTimeOut() != null) {
+            schemaNamesCache = EvictableCacheBuilder.newBuilder().expireAfterWrite(Duration.ofMillis(config.getCacheTimeOut().toMillis())).build();
+            tableNamesCache = EvictableCacheBuilder.newBuilder().expireAfterWrite(Duration.ofMillis(config.getCacheTimeOut().toMillis())).build();
+            tableMetaCache = EvictableCacheBuilder.newBuilder().expireAfterWrite(Duration.ofMillis(config.getCacheTimeOut().toMillis())).build();
+        }
+    }
+
+    public InfluxConfig getConfig()
+    {
+        return config;
+    }
+
+    public InfluxOptimizeConfig getOptimizeConfig()
+    {
+        return optimizeConfig;
+    }
+
+    private static InfluxDB createClient(InfluxConfig config, OkHttpClient.Builder httpClient)
+    {
+        return InfluxDBFactory.connect(config.getEndpoint(), config.getUsername(), config.getPassword(), httpClient);
+    }
+
+    public void close()
+    {
+        client.close();
+    }
+
+    public String version()
+    {
+        return client.version();
+    }
+
+    public void write(String database, InfluxData influxData)
+    {
+        client.setDatabase(database);
+        Point point = Point.measurement(influxData.getMeasurement())
+                .time(influxData.getTime(), influxData.getPrecision())
+                .tag(influxData.getTags())
+                .fields(influxData.getFields())
+                .build();
+        client.write(point);
+    }
+
+    public void execute(String command)
+    {
+        Query query = new Query(command);
+        client.query(query);
```

Can we throw an exception if `QueryResult` has an error?